src/components/ha-base-time-input.ts

```diff
@@ -249,13 +253,28 @@ export class HaBaseTimeInput extends LitElement {
               <mwc-list-item value="AM">AM</mwc-list-item>
               <mwc-list-item value="PM">PM</mwc-list-item>
             </ha-select>`}
+        ${this.clearable && !this.required && !this.disabled
+          ? html`<ha-icon-button
+              label="clear"
+              @click=${this._clearValue}
+              .path=${mdiClose}
+            ></ha-icon-button>`
+          : nothing}
       </div>
       ${this.helper
         ? html`<ha-input-helper-text>${this.helper}</ha-input-helper-text>`
         : ""}
     `;
   }
 
+  private _clearValue(): void {
+    const value = undefined;
+
+    fireEvent(this, "value-changed", {
+      value,
```

Comment:
This is not allowed according to the current typing (that should have been checked here)

Reply:
It looks like it is not necessarily a `TimeChangedEvent`
![image](https://github.com/home-assistant/frontend/assets/13403863/0f9492b9-c537-4187-9c9a-00724b2955e0)

Reply:
I changed the typing of the `_timeChanged` method👍